### ISSUE TITLE
Always end logged message with a newline?

### ISSUE THREAD

**Issue by viccuad** [Masterminds/log-go#1]
Hi Matt. Nice logging library :).

The standard [log](https://golang.org/pkg/log/) Go package ensures that each logged message ends with a newline. Would it make sense to have the same behaviour here, for consistency? Perhaps we want to make implementations composable, to append functionality to the log output, so it would need to ensure that there's only 1, for example.

**Comment by mattfarina** [comment]
Thanks for bringing this up. Looking at the things I discovered I needed to fix newline handling in the cli package.

I think it's important for us to think about what we need to accomplish. Do we want to produce a new logging package that can connect with all kinds of back-end writers? If that is what I thought we needed then I would have just gone with https://github.com/apex/log. apex logging and logrus are each really slow systems for logging when it comes to performance. If you combine them together, which does not fit nicely, it will be immensely slow.

We need a logging setup that lets us write to the console (sometimes with a TTY) and also write to logrus. logrus is in maintenance mode rather than active development mode so we may someday need to replace it with something else.

This lends itself to being an interface with implementations. It's how many other languages already work. If we put a layer in there to ensure a newline at the end we are building something that's more and more complicated than an interface based system. Do we really need that?
